Sources/Smithy/Logging/SwiftLog+LogAgent.swift

```diff
@@ -56,23 +42,3 @@ public struct SwiftLogger: LogAgent {
         )
     }
 }
-
-extension LogAgentLevel {
```

Comment:
I'm not sure that we want to get rid of this enum and use `swift-log`'s levels directly.  Better to keep the third-party library isolated to a small set of files that are used to 

Also, this is a significant breaking change to our public APIs for logging.  I'm fairly certain this will break Amplify-swift, for instance.

Reply:
The log levels in `swift-log` were set [5 yrs ago and haven't changed since](https://github.com/apple/swift-log/blame/96a2f8a0fa41e9e09af4585e2724c4e825410b91/Sources/Logging/Logging.swift#L847-L877), so I think it should be safe to use and get rid of the exact same enum with same values and conversion utility method on our side.

Re: breaking change, yeah I do see that `LogAgentLevel` is used by `amplify-swift` in [a couple places](https://github.com/search?q=repo%3Aaws-amplify%2Famplify-swift%20LogAgentLevel&type=code). But they updated their usage of SDK logging for the previous [`@autoclosure` changes as a chore commit](https://github.com/aws-amplify/amplify-swift/commit/152882eb4236c8ccc3eb6cd6008a2beb2c4b0c6e) so it shouldn't be a lot of work to update to using `Logger.Level` directly.

Reply:
Moving to `@autoclosure` only broke test code, i.e. mock loggers, because the compiler takes care of converting a value to an autoclosure at the call site.  Changing a param to a whole different type is a much more obtrusive change that affects customer production code.

Reply:
Reverted.